backend/fms_core/migrations/0065_v4_12_0.py

```diff
@@ -1,43 +1,64 @@
 
-    PropertyType = apps.get_model("fms_core", "PropertyType")
-    Protocol = apps.get_model("fms_core", "Protocol")
-    PropertyType = apps.get_model("fms_core", "PropertyType")
-    ContentType = apps.get_model('contenttypes', 'ContentType')
+def add_sample_qc_distinction_dna_rna(self, apps, schema_editor):
 
-    protocol_content_type = ContentType.objects.get_for_model(Protocol)
+    STEPS = [
+        {"name": "Sample QC (DNA)", "protocol_name": "Sample QC", "specifications": [
+            {"display_name": "Sample QC Type", "sheet_name": "SampleQC", "column_name": "Sample Kind", "value": "DNA"}]
+        },
+        {"name": "Sample QC (RNA)", "protocol_name": "Sample QC", "specifications": [
+            {"display_name": "Sample QC Type", "sheet_name": "SampleQC", "column_name": "Sample Kind", "value": "RNA"}]
+        }
+    ]
 
     with reversion.create_revision(manage_manually=True):
         admin_user = User.objects.get(username=ADMIN_USERNAME)
-        
-        reversion.set_comment(f"Set '{PROPERTY_TYPE_NAME}' property types as optional.")
+        reversion.set_comment(f"Create Sample QC DNA and RNA step and transfer existing Sample QC data to their respective new step.")
         reversion.set_user(admin_user)
-
-        for protocol_name in PROTOCOL_NAMES:
-            object_id = Protocol.objects.get(name=protocol_name).id
-            property_type = PropertyType.objects.get(name=PROPERTY_TYPE_NAME, object_id=object_id, content_type=protocol_content_type)
-            property_type.is_optional = True
-            property_type.save()
-            reversion.add_to_revision(property_type)
+        for step_info in STEPS:
+            protocol = Protocol.objects.get(name=step_info["protocol_name"])
+            step = Step.objects.create(name=step_info["name"],
+                                       protocol=protocol,
+                                       created_by_id=admin_user.id,
+                                       updated_by_id=admin_user.id)
+            reversion.add_to_revision(step)
+            for specification in step_info["specifications"]:
+                step_specification = StepSpecification.objects.create(display_name=specification["display_name"],
+                                                                      sheet_name=specification["sheet_name"],
+                                                                      column_name=specification["column_name"],
+                                                                      value=specification["value"],
+                                                                      step=step,
+                                                                      created_by_id=admin_user.id,
+                                                                      updated_by_id=admin_user.id)
+                reversion.add_to_revision(step_specification)
+        oldStep = Step.objects.get(name="Sample QC")
+        if oldStep:
```

Comment:
don't test for this if it does not exist we want that migration to fail ... something is wrong.